virtual-scroller-element.js

```diff
@@ -1,18 +1,22 @@
+import {_item, _key, ItemSource} from './item-source.js';
```

Comment:
It feels weird to import these symbols here, all we need here is to check if the provided itemSource implements `item(), getLength(), key()` - maybe that can be an utility method?

Reply:
The ItemSource doesn't expose those as public APIs. Those are capabilities that the ItemSource encapsulates, but we don't want to call the public methods of the ItemSource; in general classes in the web platform talk to each other through their internals, not their public methods.

Reply:
Uhm, I see. I like it!
So if I wanted to make those methods public, I'd do something like this, right?
```js
import {ItemSource, _key, _item} from 'item-source.js';
class ContactSource extends ItemSource {
  get key() {
    return this[_key];
  }
  get item() {
    return this[_item];
  }
}
```

Reply:
Well, I don't think we'd expose the _key and _item to web developers. This is more just me being conservative until we know what would be useful for web developers: if we have use cases where item() or key() are useful, we should just expose them directly on ItemSource. (Also, I'd probably tweak things to use a proxy so that instead of source.item(3) you could just do source[3].)